eradiate/radprops/rad_profile.py

```diff
@@ -666,10 +580,10 @@ def to_dataset(self):
         """
         return make_dataset(
             wavelength=self._wavelength,
-            z_level=self._thermoprops.z_level.values,
-            z_layer=self._thermoprops.z_layer.values,
-            sigma_a=self.sigma_a.flatten(),
-            sigma_s=self.sigma_s.flatten(),
+            z_level=self.eval_thermoprops_profile().z_level.values,
+            z_layer=self.eval_thermoprops_profile().z_layer.values,
```

Comment:
Isn't it more efficient to create the `thermoprops` object once and call `make_dataset()` with `z_level = thermoprops.z_level.values` and `z_layer = thermoprops.z_layer.values`?

Reply:
Good point, I'll optimise that.